src/server/bitops_family.cc

```diff
@@ -181,60 +253,237 @@ bool SetBitValue(uint32_t offset, bool bit_value, std::string* entry) {
+      std::string default_str{values[max_len_index]};
+      return BitOpString(OrOp, SkipOr, std::move(values), std::move(default_str));
+    } else if (op == XOR_OP_NAME) {
+      return BitOpString(XorOp, SkipXor, std::move(values), std::string(max_len, 0));
+    } else if (op == AND_OP_NAME) {
+      return BitOpString(AndOp, SkipAnd, std::move(values), std::string(max_len, 0));
+    } else if (op == NOT_OP_NAME) {
+      return BitOpNotString(values[0]);
+    } else {
+      LOG(FATAL) << "Operation not supported '" << op << "'";
+      return std::string{};  // otherwise we will have warning of not returning value
+    }
+  };
+
+  if (values.empty()) {  // this is ok in case we don't have the src keys
+    return std::string{};
+  }
+  // The new result is the max length input
+  max_len = values[0].size();
+  for (std::size_t i = 1; i < values.size(); ++i) {
+    if (values[i].size() > max_len) {
+      max_len = values[i].size();
+      max_len_index = i;
+    }
+  }
+  return BitOperation();
+}
+
+OpResult<std::string> CombineResultOp(ShardStringResults result, std::string_view op) {
+  // take valid result for each shard
+  BitsStrVec values;
+  int index = 0;
+  for (auto&& res : result) {
+    if (res) {
+      auto v = res.value();
+      values.emplace_back(std::move(v));
+    } else {
+      if (res.status() != OpStatus::SKIPPED) {  // something went wrong, just bale out
+        return res;
+      }
+      ++index;
+    }
+  }
+  // and combine them to single result
+  return RunBitOperationOnValues(op, values);
+}
+
+// For bitop not - we cannot accumulate
+OpResult<std::string> RunBitOpNot(const OpArgs& op_args, ArgSlice keys) {
+  DCHECK(keys.size() == 1);
+
+  EngineShard* es = op_args.shard;
+  // if we found the value, just return, if not found then skip, otherwise report an error
+  auto key = keys.front();
+  OpResult<PrimeIterator> find_res = es->db_slice().Find(op_args.db_cntx, key, OBJ_STRING);
+  if (find_res) {
+    return GetString(es, find_res.value()->second);
+  } else {
+    if (find_res.status() == OpStatus::KEY_NOTFOUND) {
+      return OpStatus::SKIPPED;
+    } else {
+      return find_res.status();
+    }
+  }
+}
+// Read only operation where we are running hte bit operation on all the
+// values that belong to same shard.
+OpResult<std::string> RunBitOpOnShard(std::string_view op, const OpArgs& op_args, ArgSlice keys) {
+  DCHECK(!keys.empty());
+  if (op == NOT_OP_NAME) {
+    return RunBitOpNot(op_args, keys);
+  }
+  EngineShard* es = op_args.shard;
+  BitsStrVec values;
+  values.reserve(keys.size());
+
+  // collect all the value for this shard
+  for (auto& key : keys) {
+    OpResult<PrimeIterator> find_res = es->db_slice().Find(op_args.db_cntx, key, OBJ_STRING);
+    if (find_res) {
+      values.emplace_back(std::move(GetString(es, find_res.value()->second)));
+    } else {
+      if (find_res.status() ==
+          OpStatus::KEY_NOTFOUND) {  // this is allowed, just return empty string per Redis
```

Comment:
you check here for `KEY_NOTFOUND` and return empty string.
at line 425 you check for `KEY_NOTFOUND` and replace with SKIPPED. and then at line 404 you check for skipped.

seems like lack of consistency

Reply:
change this so that it will just skip in this case and not save any empty string